libcontainer/cgroups/systemd/common.go

```diff
@@ -28,9 +29,11 @@ const (
 )
 
 var (
-	connOnce sync.Once
-	connDbus *systemdDbus.Conn
-	connErr  error
+	isRootless bool
+	isRenewing int32
+	connOnce   sync.Once
+	connDbus   *systemdDbus.Conn
+	connErr    error
```

Comment:
Honestly, the longer I look at this the less convinced I am that we really need to make this global. How much performance are we really getting from stashing this in a global, and if we add locks I imagine this will only get less performant (not to mention more complicated).

I looked at the git history and it seems like this has always been global, so my guess is that it was written that way because it seemed right at the time -- I would suggest we remove this and make it local and not worry about creating the connection a few more times.

Reply:
Yeah this logic came from pre-libcontainer docker (3ce347c35fd45aab9da0c3f94c5c4b4e0bc49d44) and I'm not sure it's worth looking into it much further.

Reply:
Okay, it has always been implemented this way (moby/moby@6c7835050e53b733181ddfca6152c358fd625400). @alexlarsson, I know it's been a while but is there a particular reason you implemented it this way -- would it be okay to make it non-global?